common/tls/src/main/java/io/helidon/common/tls/ConfiguredTlsManager.java

```diff
@@ -91,301 +87,154 @@ public String type() {
-                                             new X509KeyManager[] {target.manager().get().keyManager().get()});
-            } else {
-                kmData = null;
-            }
-            if (kmData != null) {
-                km = kmData.result();
-                keyManager.accept(kmData.original());
-            }
+            SSLContext sslContext;
 
-            ReloadableData<TrustManager, X509TrustManager> tmData;
-            if (tmf != null) {
-                tmData = wrapX509TrustManagers(reloadable, tmf.getTrustManagers());
-            } else if (target.enabled()
-                    && target.manager().isPresent()
-                    && target.manager().get().trustManager().isPresent()) {
-                tmData = wrapX509TrustManagers(reloadable,
-                                               new X509TrustManager[] {target.manager().get().trustManager().get()});
+            if (tlsConfig.provider().isPresent()) {
+                sslContext = SSLContext.getInstance(tlsConfig.protocol(), tlsConfig.provider().get());
             } else {
-                tmData = null;
+                sslContext = SSLContext.getInstance(tlsConfig.protocol());
             }
-            if (tmData != null) {
-                tm = tmData.result();
-                trustManager.accept(tmData.original());
-            }
-
             sslContext.init(km, tm, secureRandom);
 
             SSLSessionContext serverSessionContext = sslContext.getServerSessionContext();
             if (serverSessionContext != null) {
-                serverSessionContext.setSessionCacheSize(target.sessionCacheSize());
+                serverSessionContext.setSessionCacheSize(tlsConfig.sessionCacheSize());
                 // seconds
-                serverSessionContext.setSessionTimeout((int) target.sessionTimeout().toSeconds());
+                serverSessionContext.setSessionTimeout((int) tlsConfig.sessionTimeout().toSeconds());
             }
 
-            target.sslContext(sslContext);
-            return sslContext;
+            this.sslContext = sslContext;
         } catch (GeneralSecurityException e) {
             throw new IllegalArgumentException("Failed to create SSLContext", e);
         }
     }
 
-    static TrustManagerFactory buildTrustAllTmf() {
-        return new TrustAllManagerFactory();
-    }
-
-    static TrustManagerFactory buildTmf(TlsConfig.BuilderBase<?, ?> target) throws KeyStoreException {
-        KeyStore ks = keystore(target);
-        int i = 1;
-        for (X509Certificate cert : target.trust()) {
-            ks.setCertificateEntry(String.valueOf(i), cert);
-            i++;
-        }
-        TrustManagerFactory tmf = tmf(target);
-        tmf.init(ks);
-        return tmf;
-    }
-
-    static SecureRandom secureRandom(TlsConfig.BuilderBase<?, ?> target)
-            throws NoSuchAlgorithmException, NoSuchProviderException {
-        if (target.secureRandom().isPresent()) {
```

Comment:
Also note that besides SSLContext, the secureRandom was capable of being set by the builder/decorator.  Seems like we are losing that here too if managers are not decorators.  With this new approach the secureRandom seems to be produced via `getInstance()` each time.